airbyte_cdk/sources/declarative/concurrent_declarative_source.py

```diff
@@ -193,6 +201,17 @@ def _group_streams(
                     declarative_stream.name
                 ].get("incremental_sync")
 
+                partition_router_component_definition = (
+                    name_to_stream_mapping[declarative_stream.name]
+                    .get("retriever")
+                    .get("partition_router")
+                )
+
+                is_substream_without_incremental = (
+                    partition_router_component_definition
+                    and not datetime_based_cursor_component_definition
```

Comment:
nit: The naming makes it a big confusing. Should we rename `datetime_based_cursor_component_definition` for `incremental_sync` to clarify that it can be custom incremental syncs?